File: CS/StyleFormatConditionsForXtraVerticalGrid/CustomVGridControl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraVerticalGrid;
using DevExpress.XtraGrid;
using System.ComponentModel;
using DevExpress.XtraVerticalGrid.Rows;
using DevExpress.XtraVerticalGrid.Data;

namespace StyleFormatConditionsForXtraVerticalGrid {
    class CustomVGridControl:VGridControl {
        MyStyleFormatConditionCollection formatConditions;
        public CustomVGridControl() {
            this.formatConditions = new MyStyleFormatConditionCollection(this);
        }

        public VGridDataManager DataManager { get { return base.DataManager; } }

        private void FillCellCollectionStyle() {
            for(int i=0; i < this.RecordCount; i++)
                cellStyleCollection.Add(i,false);
        }
       
        Dictionary<int, bool> cellStyleCollection;
        protected override void RaiseRecordCellStyle(DevExpress.XtraVerticalGrid.Events.GetCustomRowCellStyleEventArgs e) {
            if (cellStyleCollection == null) {
                cellStyleCollection = new Dictionary<int, bool>();
                FillCellCollectionStyle();
            }
            int conditionsCount = FormatConditions.Count;
            if (conditionsCount == 0) return;

            for (int n = 0; n < conditionsCount; n++) {
                MyStyleFormatCondition cond = FormatConditions[n] as MyStyleFormatCondition;
                bool isSatisfyCondition = CalcCondition(e.Row, e.RecordIndex, cond);

                if (isSatisfyCondition)
                    cellStyleCollection[e.RecordIndex] = isSatisfyCondition;

                if (cond.ApplyToRow) {
                    if (cellStyleCollection[e.RecordIndex]) {
                        e.Appearance = cond.Appearance;
                    }

                }
                else if (!cond.ApplyToRow && isSatisfyCondition) {
                    e.Appearance = cond.Appearance;
                }

            }
        }


        private bool CalcCondition(BaseRow row, int recordIndex, MyStyleFormatCondition cond) {
            if (!cond.IsValid) return false;
            
            return cond.CheckValue(row, GetCellValue(row, recordIndex), recordIndex);
        }


        public MyStyleFormatConditionCollection FormatConditions { get { return formatConditions; } }
   
    }
   
}

File: VB/StyleFormatConditionsForXtraVerticalGrid/CustomVGridControl.vb
' Developer Express Code Central Example:
' How to implement StyleFormatConditions in VGridControl
' 
' This example demonstrates how to implement StyleFormatConditions in
' VGridControl. The difficulty is in the fact that VGridControl doesn't implement
' the FormatConditions property at all, if compared to how this behavior is
' supported in GridControl. So, we created a custom class inherited from
' VGridControl and added the FormatConditions property to it. In this example we
' override the RecordCellStyle event to set a new style for VGridControl’s cells,
' which satisfy predefined conditions.
' 
' You can find sample updates and versions for different programming languages here:
' http://www.devexpress.com/example=E4106


Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraVerticalGrid
Imports DevExpress.XtraGrid
Imports System.ComponentModel
Imports DevExpress.XtraVerticalGrid.Rows
Imports DevExpress.XtraVerticalGrid.Data

Namespace StyleFormatConditionsForXtraVerticalGrid
	Friend Class CustomVGridControl
		Inherits VGridControl
		Private formatConditions_Renamed As MyStyleFormatConditionCollection
		Public Sub New()
			Me.formatConditions_Renamed = New MyStyleFormatConditionCollection(Me)
		End Sub

		Public ReadOnly Property DataManager() As VGridDataManager
			Get
				Return MyBase.DataManager
			End Get
		End Property

		Private Sub FillCellCollectionStyle()
			For i As Integer = 0 To Me.RecordCount - 1
				cellStyleCollection.Add(i,False)
			Next i
		End Sub

		Private cellStyleCollection As Dictionary(Of Integer, Boolean)
		Protected Overrides Sub RaiseRecordCellStyle(ByVal e As DevExpress.XtraVerticalGrid.Events.GetCustomRowCellStyleEventArgs)
			If cellStyleCollection Is Nothing Then
				cellStyleCollection = New Dictionary(Of Integer, Boolean)()
				FillCellCollectionStyle()
			End If
			Dim conditionsCount As Integer = FormatConditions.Count
			If conditionsCount = 0 Then
				Return
			End If

			For n As Integer = 0 To conditionsCount - 1
				Dim cond As MyStyleFormatCondition = TryCast(FormatConditions(n), MyStyleFormatCondition)
				Dim isSatisfyCondition As Boolean = CalcCondition(e.Row, e.RecordIndex, cond)

				If isSatisfyCondition Then
					cellStyleCollection(e.RecordIndex) = isSatisfyCondition
				End If


				If cond.ApplyToRow Then
					If cellStyleCollection(e.RecordIndex) Then
						e.Appearance = cond.Appearance
					End If

				ElseIf (Not cond.ApplyToRow) AndAlso isSatisfyCondition Then
					e.Appearance = cond.Appearance
				End If

			Next n
		End Sub


		Private Function CalcCondition(ByVal row As BaseRow, ByVal recordIndex As Integer, ByVal cond As MyStyleFormatCondition) As Boolean
			If (Not cond.IsValid) Then
				Return False
			End If

			Return cond.CheckValue(row, GetCellValue(row, recordIndex), recordIndex)
		End Function


		Public ReadOnly Property FormatConditions() As MyStyleFormatConditionCollection
			Get
				Return formatConditions_Renamed
			End Get
		End Property

	End Class

End Namespace
